rfcs/button-group-component.md

```diff
@@ -0,0 +1,43 @@
+# Button Group Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {ButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <ButtonGroup>
+      <Button>Copy</Button>
+      <Button>Cut</Button>
+      <Button>Paste</Button>
+    </ButtonGroup>
+}
+```
+
+## Exports
+
+* `ButtonGroup`
+
+## `ButtonGroup` API
+
+* `children: (props: Props) => React.Node` - Required
+  This can be an SVG icon as well.
+* `ariaLabel: string` - Optional:
+  Accessible label
+* `disabled: boolean` - Optional:
+  Defines if the button group disabled.
+* `size: 'default' | 'compact'`
+  Defines the size of a button
+* `onChange: func` - Optinal:
+  Called when a toggleable Button is selected
+* `onClick: func` - Optional:
+  Called with the click event
```

Comment:
So just to clarify, is this component meant to behave almost like a radio group? Or is it just meant to visually group the buttons together?

If just the latter, then maybe we don't need `onChange` and `onClick` handlers here. I'm trying to figure out what I would do with those as a customer – seems like it makes more sense to just attach onClick to each individual buttons.

But if it's the former (more like a radio group, where a button can be "selected"), then I think we may want to align more closely with the radio group API. This is what antd does: https://codesandbox.io/s/6xoqyy57o3

Which one were you thinking?

cc @nadiia @abmai @blackswanny

Reply:
oh, I see - [mineral-ui](https://mineral-ui.com/components/button-group) does something very similar.

I think based on this discussion, we should follow up with design to see what behavior they had in mind - does that sound reasonable?

Reply:
Yeah that's probably a good idea.

Reply:
Yes, let's extend the question to the design team. But I'd assume every button would need an individual click or change handler.